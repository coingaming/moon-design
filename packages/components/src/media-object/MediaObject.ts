import styled, { CSSObject } from 'styled-components';
import { themed } from '@heathmont/sportsbet-utils';
import { SpaceProps } from '@heathmont/sportsbet-themes';

export type MediaObjectProps = {
  alignItems?: CSSObject['alignItems'];
  /**
   * Excepts a string value, number or a theme key.
   *
   * Theme keys are returned as a `rem`-based unit.
   *
   * e.g. `<MediaObject gridGap="large" />` returns `space.large` as a rem unit.
   */
  gridGap?: SpaceProps | CSSObject['gridGap'];
  /**
   * Excepts a string value or a theme key.
   *
   * Theme keys are returned as a `rem`-based unit.
   *
   * e.g. `<MediaObject mediaWidth="large" />` returns `space.large` as a rem unit.
   */
  mediaWidth?: SpaceProps | string | undefined;
  reverse?: boolean;
  textAlign?: CSSObject['textAlign'];
};

const MediaObject = styled.div<MediaObjectProps>(
  ({
    alignItems = 'center',
    gridGap = 'default',
    mediaWidth = 'large',
    reverse,
    textAlign = 'left',
    theme,
  }) => [
    {
      alignItems,
      gridGap: themed('space', gridGap)(theme),
      textAlign,
      display: 'grid',
      gridTemplate: `max-content / ${themed('space', mediaWidth)(theme)} auto`,
      'img,svg': {
        display: 'block',
        width: '100%',
        height: 'auto',
      },
    },
    reverse && {
      direction: 'rtl',
      '& > *': {
        direction: 'ltr',
      },
    },
  ]
);

export default MediaObject;
