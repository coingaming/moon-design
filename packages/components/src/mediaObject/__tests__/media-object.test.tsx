import React from 'react';
import { create } from 'react-test-renderer';
import { sportsbetDark, ThemeProvider } from '@heathmont/moon-themes';
import 'jest-styled-components';

import MediaObject from '../MediaObject';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={sportsbetDark}>{component}</ThemeProvider>
);

describe('MediaObject', () => {
  test('renders with `default` settings', () => {
    const mediaObject = create(
      renderWithTheme(
        <MediaObject>
          <img alt="Example" src="https://via.placeholder.com/50C" />
          <div>
            <p>Live chat</p>
            <p>We offer 24/7 support</p>
          </div>
        </MediaObject>
      )
    );

    expect(mediaObject).toMatchSnapshot();
  });

  test('renders with custom CSS grid props', () => {
    const mediaObject = create(
      renderWithTheme(
        <React.Fragment>
          <MediaObject gridGap="2rem" alignItems="top">
            <img alt="Example" src="https://via.placeholder.com/50C" />
            <div>
              <p>Live chat</p>
              <p>We offer 24/7 support</p>
            </div>
          </MediaObject>
          <MediaObject gridGap="large" alignItems="top">
            <img alt="Example" src="https://via.placeholder.com/50C" />
            <div>
              <p>Live chat</p>
              <p>We offer 24/7 support</p>
            </div>
          </MediaObject>
        </React.Fragment>
      )
    );

    expect(mediaObject).toMatchSnapshot();
  });

  test('renders with custom media size', () => {
    const mediaObject = create(
      renderWithTheme(
        <React.Fragment>
          <MediaObject mediaWidth="40px">
            <img alt="Example" src="https://via.placeholder.com/50C" />
            <div>
              <p>Live chat</p>
              <p>We offer 24/7 support</p>
            </div>
          </MediaObject>
          <MediaObject mediaWidth="xlarge">
            <img alt="Example" src="https://via.placeholder.com/50C" />
            <div>
              <p>Live chat</p>
              <p>We offer 24/7 support</p>
            </div>
          </MediaObject>
        </React.Fragment>
      )
    );

    expect(mediaObject).toMatchSnapshot();
  });

  test('renders in reverse', () => {
    const mediaObject = create(
      renderWithTheme(
        <div>
          <MediaObject reverse>
            <img alt="Example" src="https://via.placeholder.com/50C" />
            <div>
              <p>Live chat</p>
              <p>We offer 24/7 support</p>
            </div>
          </MediaObject>

          <MediaObject reverse textAlign="right">
            <img alt="Example" src="https://via.placeholder.com/50C" />
            <div>
              <p>Live chat</p>
              <p>We offer 24/7 support</p>
            </div>
          </MediaObject>
        </div>
      )
    );

    expect(mediaObject).toMatchSnapshot();
  });
});
