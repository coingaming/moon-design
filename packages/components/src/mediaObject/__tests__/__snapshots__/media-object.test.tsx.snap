// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaObject renders in reverse 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-gap: 1rem;
  text-align: left;
  display: grid;
  grid-template: max-content / 2rem auto;
  direction: rtl;
}

.c0 img,
.c0 svg {
  display: block;
  width: 100%;
  height: auto;
}

.c0 > * {
  direction: ltr;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-gap: 1rem;
  text-align: right;
  display: grid;
  grid-template: max-content / 2rem auto;
  direction: rtl;
}

.c1 img,
.c1 svg {
  display: block;
  width: 100%;
  height: auto;
}

.c1 > * {
  direction: ltr;
}

<div>
  <div
    className="c0"
  >
    <img
      alt="Example"
      src="https://via.placeholder.com/50C"
    />
    <div>
      <p>
        Live chat
      </p>
      <p>
        We offer 24/7 support
      </p>
    </div>
  </div>
  <div
    className="c1"
  >
    <img
      alt="Example"
      src="https://via.placeholder.com/50C"
    />
    <div>
      <p>
        Live chat
      </p>
      <p>
        We offer 24/7 support
      </p>
    </div>
  </div>
</div>
`;

exports[`MediaObject renders with \`default\` settings 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-gap: 1rem;
  text-align: left;
  display: grid;
  grid-template: max-content / 2rem auto;
}

.c0 img,
.c0 svg {
  display: block;
  width: 100%;
  height: auto;
}

<div
  className="c0"
>
  <img
    alt="Example"
    src="https://via.placeholder.com/50C"
  />
  <div>
    <p>
      Live chat
    </p>
    <p>
      We offer 24/7 support
    </p>
  </div>
</div>
`;

exports[`MediaObject renders with custom CSS grid props 1`] = `
Array [
  .c0 {
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  grid-gap: 2rem;
  text-align: left;
  display: grid;
  grid-template: max-content / 2rem auto;
}

.c0 img,
.c0 svg {
  display: block;
  width: 100%;
  height: auto;
}

<div
    className="c0"
  >
    <img
      alt="Example"
      src="https://via.placeholder.com/50C"
    />
    <div>
      <p>
        Live chat
      </p>
      <p>
        We offer 24/7 support
      </p>
    </div>
  </div>,
  .c0 {
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  grid-gap: 2rem;
  text-align: left;
  display: grid;
  grid-template: max-content / 2rem auto;
}

.c0 img,
.c0 svg {
  display: block;
  width: 100%;
  height: auto;
}

<div
    className="c0"
  >
    <img
      alt="Example"
      src="https://via.placeholder.com/50C"
    />
    <div>
      <p>
        Live chat
      </p>
      <p>
        We offer 24/7 support
      </p>
    </div>
  </div>,
]
`;

exports[`MediaObject renders with custom media size 1`] = `
Array [
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-gap: 1rem;
  text-align: left;
  display: grid;
  grid-template: max-content / 40px auto;
}

.c0 img,
.c0 svg {
  display: block;
  width: 100%;
  height: auto;
}

<div
    className="c0"
  >
    <img
      alt="Example"
      src="https://via.placeholder.com/50C"
    />
    <div>
      <p>
        Live chat
      </p>
      <p>
        We offer 24/7 support
      </p>
    </div>
  </div>,
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-gap: 1rem;
  text-align: left;
  display: grid;
  grid-template: max-content / 2.5rem auto;
}

.c0 img,
.c0 svg {
  display: block;
  width: 100%;
  height: auto;
}

<div
    className="c0"
  >
    <img
      alt="Example"
      src="https://via.placeholder.com/50C"
    />
    <div>
      <p>
        Live chat
      </p>
      <p>
        We offer 24/7 support
      </p>
    </div>
  </div>,
]
`;
